Stratis.Bitcoin.Features.Consensus/ConsensusLoop.cs

```diff
@@ -1,47 +1,146 @@
 ﻿using Microsoft.Extensions.Logging;
 using NBitcoin;
+using NBitcoin.Protocol;
+using Stratis.Bitcoin.Base;
 using Stratis.Bitcoin.Base.Deployments;
 using Stratis.Bitcoin.BlockPulling;
+using Stratis.Bitcoin.Connection;
 using Stratis.Bitcoin.Features.Consensus.CoinViews;
 using Stratis.Bitcoin.Utilities;
+using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
 
 namespace Stratis.Bitcoin.Features.Consensus
 {
-    public class BlockResult
+    /// <summary>
+    /// A block object that is passed around when downloading a block from peers and has information about the validation state of the block.
+    /// </summary>
+    public class BlockItem
```

Comment:
I would be up for another name for this class, perhaps ConsensusBlock or ValidationBlock

Reply:
sure, no problem with that, just make sure you are aware of all code that actually uses it, so that the name fits all